.circleci/config.yml

```diff
@@ -1,84 +1,48 @@
-          keys:
-            - yarn-{{ checksum "yarn.lock" }}
-            - yarn-
-
+      - setup_remote_docker
       - run:
-          name: Yarn install
-          command: yarn --check-files
-
-      - save_cache:
-          name: Save yarn cache
-          key: yarn-{{ checksum "yarn.lock" }}
-          paths:
-            - node_modules/
-
-      - run:
-          name: Poll for Elasticsearch
-          command: sleep 10 && curl --retry 10 --retry-delay 5 -v http://127.0.0.1:9200/
-
-      - run:
-          name: Yarn test
-          command: yarn test
-
+          name: Test
+          command: COMMIT_HASH=$(git rev-parse --short HEAD) hokusai test --build
   deploy-staging:
     docker:
-      - image: circleci/node:10-stretch-browsers
-
+      - image: artsy/hokusai
     steps:
+      - add_ssh_keys
       - checkout
-
       - run:
-          name: Yarn assets
-          command: |
-            curl https://cli-assets.heroku.com/install-ubuntu.sh | sh
-            bash .circleci/setup_heroku.sh
-            yarn install
-            yarn assets
-            yarn bucket-assets -b positron-staging
-            heroku config:set ASSET_MANIFEST=$(cat manifest.json) --app=positron-staging
-
+          name: Configure
+          command: hokusai configure --kubectl-version 1.10.7 --s3-bucket artsy-citadel --s3-key k8s/config --platform linux
+      - run:
+          name: Push
+          command: COMMIT_HASH=$(git rev-parse --short HEAD) hokusai registry push --tag $CIRCLE_SHA1 --force --overwrite
       - run:
           name: Deploy to staging
-          command: git push --force https://heroku:$HEROKU_TOKEN@git.heroku.com/positron-staging.git master
-
+          command: hokusai staging deploy $CIRCLE_SHA1
       - run:
           name: Update staging git branch
           command: git push git@github.com:artsy/positron.git $CIRCLE_SHA1:staging --force
-
   deploy-production:
     docker:
-      - image: circleci/node:10-stretch-browsers
-
+      - image: artsy/hokusai
     steps:
+      - add_ssh_keys
       - checkout
-
       - run:
-          name: Yarn assets
-          command: |
-            curl https://cli-assets.heroku.com/install-ubuntu.sh | sh
-            bash .circleci/setup_heroku.sh
-            yarn install
-            yarn assets
-            yarn bucket-assets -b positron-production
-            heroku config:set ASSET_MANIFEST=$(cat manifest.json) --app=positron-production
-
+          name: Configure
+          command: hokusai configure --kubectl-version 1.10.7 --s3-bucket artsy-citadel --s3-key k8s/config --platform linux
+      - run:
+          name: Push
+          command: COMMIT_HASH=$(git rev-parse --short HEAD) hokusai registry push --tag $CIRCLE_SHA1 --force --overwrite
       - run:
           name: Deploy to production
-          command: git push --force https://heroku:$HEROKU_TOKEN@git.heroku.com/positron-production.git master
-
+          command: hokusai production deploy $CIRCLE_SHA1
 workflows:
```

Comment:
Love how short this gets 👍

Reply:
@izakp When do you typically decide between a `hokusai pipeline promote` versus a direct deploy via `hokusai env deploy`? Are most CI-based deployments direct deploy?

Reply:
@dleve123 it's pretty much up to the project maintainers whether they prefer to deploy production via a `hokusai pipeline promote` or a PR from say, `staging` -> `release` here then have CI deploy from `release`.  Promoting will ensure the exact same image is deployed to production, while the release-branch strategy creates an additional image build via a merge commit.  @joeyAghion and I tried to decide on on good / recommended solution but ran out of ideas when we realized we couldn't set up PRs that always rebase rather than merge, which would have provided a best-of-both-worlds solution.

Reply:
There was also the additional concern that promotion could encounter a race condition if `master` was (unknown to the promoting developer) in the process of being deployed while they ran the `hokusai pipeline promote` command.